src/screens/MainApp/DiagnoseRequest/screens/ImageReview/index.js

```diff
@@ -1,28 +1,45 @@
 import React, { useState, useEffect } from 'react'
 import { View } from 'react-native'
 import Icon from 'react-native-vector-icons/FontAwesome5'
-import { Button } from '~/components'
+import { Button, ImageSelection } from '~/components'
 import NavigationService from '~/navigationService'
+import * as ImageService from '~/imageService'
 import ConfirmButton from './ConfirmButton'
 import Carousel from './Carousel'
 import ImageList from './ImageList'
-import { requestImages, newImages } from '~/constants/mocks'
-import { MIN_IMAGES } from './constants'
+import { MIN_IMAGES, MAX_IMAGES } from './constants'
 
 const ImageVisualization = (props) => {
   const [activeIndex, setActiveIndex] = useState(0)
   const [submitError, setSubmitError] = useState(false)
-  const [images, setImages] = useState(requestImages.slice())
+  const [images, setImages] = useState(props.navigation.state.params.images)
+  const [showImageSelection, setShowImageSelection] = useState(false)
 
-  const confirmRequest = () => {}
+  useEffect(() => {
+    setImages(props.navigation.state.params.images)
+  }, [props.navigation.state.params.images])
```

Comment:
You're using this chain of accessors three times. I need to change three places if I want to change the way I receive the images. Consider using a variable for this:
```javascript
const imagesFromPreviousStep = props.navigation.state.params.images

// use imagesFromPreviousStep
```